src/App.jsx

```diff
@@ -1,36 +1,85 @@
 import { useEffect, useState } from "react";
 import RestaurantButton from "./Components/RestaurantButton/index.jsx";
 import Hero from "./Components/Hero/index.jsx";
+import MenuItems from "./Components/MenuItems/index.jsx";
+import ReturnButton from "./Components/ReturnButton/index.jsx";
 
 function App() {
-  const [restaurantNames, setRestaurantNames] = useState([]);
-
+  const [restaurantInfo, setRestaurantInfo] = useState([]);
+  const [currentId, setCurrentId] = useState(0);
+  const [currentName, setCurrentName] = useState("");
+  
+  let xlMediaColumn;
+  if (currentId) { xlMediaColumn = "xl:grid-cols-6"};
 
   useEffect(() => {
-    fetch("https://food-delivery-api.dev.io-academy.uk/restaurants")
-      .then((response) => response.json())
-      .then((data) => {
-        setRestaurantNames(data);
-      });
-  }, []);
+    if (!currentId) {
+      fetch("https://food-delivery-api.dev.io-academy.uk/restaurants")
+        .then((response) => response.json())
+        .then((data) => {
+          setRestaurantInfo(data);
+        });
+    } else {
+      fetch(
+        `https://food-delivery-api.dev.io-academy.uk/restaurants/${currentId}`
+      )
+        .then((response) => response.json())
+        .then((data) => {
+          setRestaurantInfo(data.foodItems);
+          setCurrentName(data.restaurant);
```

Comment:
Why not set the state to the whole object that comes back? Then you only have one state you can pull this array and the name from.